tests/test_qaoa.py

```diff
@@ -74,11 +74,18 @@
 b_rx.add_edges_from([(0, 1, ""), (1, 2, ""), (0, 2, "")])
 
 
+def _get_ops(op):
+    if isinstance(op, qml.operation.Tensor):
+        return op.name
+    return [op.name]
+
+
 def decompose_hamiltonian(hamiltonian):
     coeffs = list(qml.math.toarray(hamiltonian.coeffs))
-    ops = [i.name for i in hamiltonian.ops]
-    wires = [i.wires for i in hamiltonian.ops]
-    return [coeffs, ops, wires]
+    # Terms in the Hamiltonian are ops acting on different wires, so the order shouldn't
+    # matter as long as the correct ops are on the correct wires.
+    wire_ops = [dict((w, c) for w, c in zip(op.wires, _get_ops(op))) for op in hamiltonian.ops]
```

Comment:
If `op` is a `Prod` with two or more ops, `_get_ops` will return `["Prod"]`, and `op.wires` will have two or more items, so the `zip` will only have one item as the iterator stops after the shorter list runs out of items.

Not a problem if we expect `decompose_hamiltonian` to only be used with legacy op math though.

Reply:
Actually `decompose_hamiltonian` is only used once ever. I can probably remove it completely and update the only test case that is still using it. This was the plan to begin with, I probably just missed it.

Reply:
I think that makes the most sense.